modin/test/storage_formats/base/test_internals.py

```diff
@@ -20,9 +20,18 @@
     df_equals,
 )
 from modin.config import NPartitions
+from modin.utils import get_current_execution
 
 NPartitions.put(4)
 
+# Our configuration in pytest.ini requires that we explicitly catch all
+# instances of defaulting to pandas, but we should always default to pandas for
+# BaseOnPython.
+if get_current_execution() == "BaseOnPython":
+    pytestmark = pytest.mark.filterwarnings(
+        "default:.*defaulting to pandas.*:UserWarning"
+    )
+
```

Comment:
Since this logic has to be located in every test file, we should probably put it into some single place, so we won't dupe the same code all over the tests and won't hurt ourselves when the warnings pattern would change making us change it everywhere.

We probably can put these lines to [`conftest.py`](https://github.com/modin-project/modin/blob/master/modin/conftest.py) so this mark applied to all of the tests automatically on the `pytest` initialization stage, will that work?

Reply:
I want the default to be that the default warnings are treated as errors, so I don't want to apply this mark to all tests automatically. Ideally we will eventually catch all the warnings and get rid of the statements that ignore them for most tests, as I've written in the TODOs. However, the `BaseOnPython` exceptions should stay because we expect to default to Pandas.

It doesn't work to set `pytestmark` in a util function in a separate file, as it only applies to the module where the definition occurs (I tried this and it didn't work). In my latest commit, I've instead defined a common variable for the `filterwarnings` string so that we should only have to update that variable whenever the warning message changes.